packages/react/src/components/button/button.tsx

```diff
@@ -1,35 +1,41 @@
 import { ButtonProps as BaseProps, c, classy, m } from '@onfido/castor';
 import { useField } from '@onfido/castor-react';
 import React, { HTMLAttributes } from 'react';
+import { withRef } from '../../utils';
 
-export const Button: ButtonComponent = ({
-  kind = 'action',
-  variant = 'primary',
-  className,
-  ...restProps
-}: ButtonProps<'a'> | ButtonProps<'button'>) => {
+export const Button: ButtonComponent = withRef(function Button(
+  {
+    kind = 'action',
+    variant = 'primary',
+    className,
+    ...restProps
+  }: ButtonProps<'a'> | ButtonProps<'button'>,
+  ref?: AnchorRef | ButtonRef
+) {
   const { disabled } = useField();
 
   const Element = (restProps as ButtonProps<'a'>).href ? 'a' : 'button';
 
   return (
     <Element
-      {...(Element === 'button' && {
-        disabled, // will be overriden by props if set
-      })}
+      // will be overriden by props if set
```

Comment:
[Minor] I prefer the old style of comment, as to me it is more clear what it targets.

Reply:
Sure, reverted.